src/table/operations.jl

```diff
@@ -0,0 +1,85 @@
+import Base: filter, map, reduce
+
+"""
+    function map(f, d::DTableRows)
+
+Applies `f` to each row of the `DTable`.
+
+# Examples
+```jldoctest        
+    d = DTable((a=rand(1000), b=rand(1000));chunksize=100)
+    m = map(x -> x.a + x.b, eachrow(d))
+    fetch(m)
+```
+"""
+function map(f, d::DTableRows)
+    chunk_wrap = chunk -> begin 
+        i = eachrow(chunk)
+        map(f, i)    
+    end
+    result_vector = [Dagger.@spawn chunk_wrap(c) for c in d.dtable.chunks]
+    Dagger.@spawn ((r...) -> vcat((r)...))(result_vector...)
+end
+
+function map(f, d::DTableColumns)
+    throw("Not implemented")
+end
+
+"""
+    function reduce(f, d::DTableRows; init)
+
+Reduces the `DTable` using function `f`.
+
+# Examples
+```jldoctest        
+    d = DTable((a=rand(1000), b=rand(1000));chunksize=100)
+    r = reduce((acc, x) -> acc + x.a, eachrow(d); init=0.0)
+    fetch(r)
+```
+"""
+function reduce(f, d::DTableRows; init)
+    chunk_wrap = chunk -> begin
+        i = eachrow(chunk)
+        reduce(f, i; init)
+    end
+   
+    chunk_results = [Dagger.@spawn chunk_wrap(c) for c in d.dtable.chunks]
+    
+    # this is dumb, probably a better way to do this
+    # it gets the function used to aggregate x and y in a reduction function 
+    # eg. for: (x,y) -> x + y; this thing will return Base.+ 
+    get_aggregate_fun = (fun) -> begin
+        ast_f = Base.uncompressed_ast(methods(fun).ms[1])
+        return_line = ast_f.code[end]
+        if hasproperty(return_line, :val) 
+            ast_id_of_ret_val = return_line.val.id # julia 1.6+
+        else
+            ast_id_of_ret_val= return_line.args[1].id # julia 1.5
+        end
+        ast_agg_op = ast_f.code[ast_id_of_ret_val].args[1]
+        getfield(ast_agg_op.mod, ast_agg_op.name)
+    end
```

Comment:
Yeah, this is no good. What's the reasoning for this?

Reply:
Well, if I assume the format of the user provided function to be 
`(acc, row) -> any_associative_fun(acc, any_user_transform(row))`
then I would want to figure out the `any_associative_fun` out of it to reduce the values obtained from reducing chunks.

I've been thinking about it all morning, but I think I've answered my question right as I was writing this answer.
I think instead of asking the user to provide a generic reduce function I'd rather ask them to provide `any_associative_fun` and `any_user_transform` separately. That would let me avoid this AST mess

I'll try that instead

Reply:
Replaced this monstrosity with an interface change to provide two separate functions